bluetooth: controller: Allow out-of-tree vendor specific HCI commands

mtpr-ot opened the pull request:
Renamed function vendor_cmd_handle to vendor_cmd_handle_common for
shared vendor commands. This allows vendor to implement vendor_cmd_handle
containing both common and specific handling.

Signed-off-by: Morten Priess <mtpr@oticon.com>

mtpr-ot commented:
This replaces PR#14143

joerchan commented on subsys/bluetooth/controller/hci/hci_internal.h:
```diff
@@ -40,3 +28,8 @@ int hci_acl_handle(struct net_buf *acl, struct net_buf **evt);
 void hci_acl_encode(struct node_rx_pdu *node_rx, struct net_buf *buf);
 void hci_num_cmplt_encode(struct net_buf *buf, u16_t handle, u8_t num);
 #endif
+int vendor_cmd_handle(u16_t ocf, struct net_buf *cmd, struct net_buf **evt);
```
I think we should prefix with hci_

joerchan commented on subsys/bluetooth/controller/hci/hci_internal.h:
```diff
@@ -40,3 +28,9 @@ int hci_acl_handle(struct net_buf *acl, struct net_buf **evt);
+			  struct net_buf **evt);
+int hci_vendor_cmd_handle_common(u16_t ocf, struct net_buf *cmd,
+			     struct net_buf **evt);
+void *cmd_complete(struct net_buf **buf, u8_t plen);
```
cmd_complete and evt_create should have hci_ as well.

mtpr-ot replied:
OK

mtpr-ot committed:
bluetooth: controller: Allow out-of-tree vendor specific HCI commands

Renamed function vendor_cmd_handle to vendor_cmd_handle_common for
shared vendor commands. This allows vendor to implement
vendor_cmd_handle, containing both common and specific handling.

Signed-off-by: Morten Priess <mtpr@oticon.com>

mtpr-ot commented:
@joerchan, any chance you could push the 'approve' button on this? Thanks.

aescolar commented:
@cvinayak : Are you happy with this change?
